plugins/ecs-bridge/e2eTests/e2e_test.go

```diff
@@ -0,0 +1,335 @@
+// if the env var is not set
+func getEnvOrDefault(name string, fallback string) string {
+	val := os.Getenv(name)
+	if val == "" {
+		return fallback
+	}
+
+	return val
+}
+
+// getBridgeLink gets a handle to the bridge device
+func getBridgeLink(t *testing.T) netlink.Link {
+	bridgeLink, err := netlink.LinkByName(bridgeName)
+	require.NoError(t, err, "Unable to find bridge: %s", bridgeName)
+	_, ok := bridgeLink.(*netlink.Bridge)
+	require.True(t, ok, "Link named '%s' is not a bridge", bridgeName)
+	return bridgeLink
+}
+
+// validateBridgeAddress validates that the bridge is set up with the expected
+// IP address
+func validateBridgeAddress(t *testing.T, bridge netlink.Link) {
+	addrs, err := netlink.AddrList(bridge, netlink.FAMILY_V4)
+	require.NoError(t, err, "Unable to list the addresses of: %s", bridge.Attrs().Name)
+	addressFound := false
+	for _, addr := range addrs {
+		if addr.IPNet.String() == expectedBridgeAddress {
+			addressFound = true
+		}
+	}
+	require.True(t, addressFound, "IP address '%s' not assigned to bridge: %s",
+		expectedBridgeAddress, bridge.Attrs().Name)
+}
+
+// getVethAndVerifyLo gets the veth pair device in the namespace. It also
+// verifies that localhost interface device exists in the namespace
+func getVethAndVerifyLo(t *testing.T) (netlink.Link, bool) {
+	links, err := netlink.LinkList()
+	require.NoError(t, err, "Unable to list devices")
+	loFound := false
+	vethFound := false
+	var veth netlink.Link
+	for _, link := range links {
+		switch link.Type() {
+		case "device":
+			if link.Attrs().Name == "lo" {
+				loFound = true
+			}
+		case "veth":
+			vethFound = true
+			veth = link
+		}
+	}
+
+	require.True(t, loFound, "localhost interface not found in netns")
+	return veth, vethFound
+}
+
+// validateVethAddress validates the address of the veth device
+func validateVethAddress(t *testing.T, veth netlink.Link) {
+	addrs, err := netlink.AddrList(veth, netlink.FAMILY_V4)
+	require.NoError(t, err, "Unable to list addresses of: %s", veth.Attrs().Name)
+	addressFound := false
+	for _, addr := range addrs {
+		if addr.IPNet.String() == expectedVethAddress {
+			addressFound = true
+		}
+	}
+	require.True(t, addressFound, "IP address '%s' not associated with: %s",
+		expectedVethAddress, veth.Attrs().Name)
+}
+
+// validateRouteForVethInTargetNetNS validates that the expected route has been
+// added for the veth device in target network namespace
+func validateRouteForVethInTargetNetNS(t *testing.T, veth netlink.Link) {
+	routes, err := netlink.RouteList(veth, netlink.FAMILY_V4)
+	require.NoError(t, err, "Unable to list routes for: %s", veth.Attrs().Name)
+	routeFound := false
+	defaultRouteFound := false
+	for _, route := range routes {
+		if route.Gw == nil {
+			defaultRouteFound = true
+		} else if route.Dst.String() == dst &&
+			route.Src == nil &&
+			route.Gw.String() == expectedGateway {
+			routeFound = true
+		}
+	}
+	require.True(t, defaultRouteFound,
```

Comment:
shouldn't this be `require.False` as we don't expect the default route. Also this is same to ipv6, so I think you can use `netlink.FAMILY_ALL` instead of just `netlink.FAMILY_V4`. To be simple, there should be only one route existed which is the one we added in the configure file.